src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2486,9 +2487,50 @@ protected virtual MethodDefinition MarkMethod (MethodReference reference, Depend
 
 			EnqueueMethod (method, reason);
 
+			// All override methods should have the same annotations as their base methods (else we will produce warning IL2046.)
+			// When marking override methods with RequiresUnreferencedCode on a type annotated with DynamicallyAccessedMembers,
+			// we should only issue a warning for the base method.
+			if (origin != null && (reason.Kind != DependencyKind.DynamicallyAccessedMember || !method.IsVirtual || Annotations.GetBaseMethods (method) == null))
+				ProcessRequiresUnreferencedCode (method, (MessageOrigin) origin, reason.Kind);
```

Comment:
I don't understand why we don't want to warn for `DynamicallyAccessedMember`. The method is kept because linker thinks that it will be used via reflection - so it should warn.

Reply:
We should still warn on these. The second part of the if condition is there to make sure we warn on covariant returns (https://github.com/mono/linker/pull/1639#discussion_r539428886).

Reply:
Makes sense